lib/usb_connection.js

```diff
@@ -50,12 +53,23 @@ USB.Connection.prototype.exec = function(command) {
 
         // Format the args into something the USB protocol can understand
         command = self._processArgsForTransport(command);
-
         // Write the bash command
         proc.control.end(command);
 
         // Once the command has been written, call the callback with the resulting process
-        proc.control.once('finish', resolve.bind(self, proc));
+        proc.control.once('finish', function commandWritten() {
+          // Clear the command failed timeout
+          clearTimeout(controlTimeoutHandle);
+          // Resolve with the process
+          resolve(proc);
+        });
+
+        controlTimeoutHandle = setTimeout(function() {
```

Comment:
We need this in order to actually get an error returned from `.exec` and raised back into the discovery process.